src/EnergyPlus/PlantComponentTemperatureSources.cc

```diff
@@ -597,35 +371,166 @@ namespace PlantComponentTemperatureSources {
         //       MODIFIED       na
         //       RE-ENGINEERED  na
 
-        // Using/Aliasing
-        using DataGlobals::SecInHour;
-        using DataHVACGlobals::TimeStepSys;
-        using DataPlant::PlantLoop;
-        using FluidProperties::GetSpecificHeatGlycol;
-
-        // SUBROUTINE PARAMETER DEFINITIONS:
         static std::string const RoutineName("CalcWaterSource");
 
-        if (WaterSource(SourceNum).MassFlowRate > 0.0) {
-            WaterSource(SourceNum).OutletTemp = WaterSource(SourceNum).BoundaryTemp;
-            Real64 Cp = GetSpecificHeatGlycol(PlantLoop(WaterSource(SourceNum).Location.loopNum).FluidName,
-                                              WaterSource(SourceNum).BoundaryTemp,
-                                              PlantLoop(WaterSource(SourceNum).Location.loopNum).FluidIndex,
-                                              RoutineName);
-            WaterSource(SourceNum).HeatRate =
-                WaterSource(SourceNum).MassFlowRate * Cp * (WaterSource(SourceNum).OutletTemp - WaterSource(SourceNum).InletTemp);
-            WaterSource(SourceNum).HeatEnergy = WaterSource(SourceNum).HeatRate * TimeStepSys * SecInHour;
+        if (this->MassFlowRate > 0.0) {
+            this->OutletTemp = this->BoundaryTemp;
+            Real64 Cp = FluidProperties::GetSpecificHeatGlycol(DataPlant::PlantLoop(this->Location.loopNum).FluidName,
+                                                               this->BoundaryTemp,
+                                                               DataPlant::PlantLoop(this->Location.loopNum).FluidIndex,
+                                                               RoutineName);
+            this->HeatRate = this->MassFlowRate * Cp * (this->OutletTemp - this->InletTemp);
+            this->HeatEnergy = this->HeatRate * DataHVACGlobals::TimeStepSys * DataGlobals::SecInHour;
         } else {
-            WaterSource(SourceNum).OutletTemp = WaterSource(SourceNum).BoundaryTemp;
-            WaterSource(SourceNum).HeatRate = 0.0;
-            WaterSource(SourceNum).HeatEnergy = 0.0;
+            this->OutletTemp = this->BoundaryTemp;
+            this->HeatRate = 0.0;
+            this->HeatEnergy = 0.0;
         }
     }
 
-    void UpdateWaterSource(int const SourceNum)
+    void WaterSourceSpecs::update()
```

Comment:
I have _no_ problem with this, but I also wouldn't have minded if the single node assignment was just done inside the calculate function as well, and this function being eliminated.